svn-type-choice #8 (#57)
ietf-rats-wg/draft-ietf-rats-corim: draft-ietf-rats-corim.md

###### Security Version Number {#sec-comid-svn}

The following details the security version number (`svn`) and the minimum security version number (`min-svn`) statements.

> A security version number is used to track changes to an object (e.g., OBJECT_INSTANCE_1, OBJECT_INSTANCE_2, ...) that are security relevant.

Reviewer: I meant `OBJECT_INSTANCE_$i` as placeholders for real things.

```suggestion
A security version number is used to track changes to an object (e.g., a Secure Enclave) that are security relevant.
```

> [^issue] https://github.com/ietf-rats-wg/draft-ietf-rats-corim/issues/8

Reviewer: this should go now that we have text.

Author: fixed in branch - this thread is outdated.

> The following describes details the security version number (svn) and the minimum security version number (min-svn).

Reviewer: There seems to be one or more missing piece(s) of English language here, but I can't tell exactly what :)

* The following describes _in_ details the security [...]
* The following details the security [...]
* The following describes _the_ details _of_ the security [...]

Reviewer: Ned, if you address this we can merge.

> A security version number is used to track changes to an object that are security relevant.

Reviewer: we should precisely define what we mean by "object" here (or possibly use another, more intuitive, term)

Author: I'm open to suggestions. We use Target Environment and `environment-map` to refer to the component that is being attested. The Target Environment could have multiple measurements such as `digest` and `raw-value-group` but these measurements could be describing different objects within the Target Environment. I don't think we have terminology for the object from which a measurement is collected. I like "object" because it is neutral. Alternatives like "component", "device", "module" all have baggage. The O in OID can mean a variety of things without too much baggage. Object does however have baggage with regard to triples structure, but it seems this is obvious when in triples-map context.

Reviewer: Maybe we could provide one or more examples of objects, e.g.:
```
A security version number is used to track changes to
an object (e.g., OBJECT_INSTANCE_1, OBJECT_INSTANCE_2)
that are security relevant.
```
which avoids both defining it or finding another term.

Author: Fixed in branch - this thread is outdated.

Rollback of a security relevant change is considered to be an attack vector, as such, svn values can't be decremented.
If a security relevant flaw is discovered in an object, the object is updated with a fix and the svn value is incremented.

> There may be several versions of an object that are in production use as a given time.

Reviewer: ```suggestion
There may be several versions of an object that are in production use at a given time.
```

If there are multiple versions having different svn values, the object with a lower svn value may

> or may not be in a security critical condition. The Endorser may specificy a minimum svn value (min-svn)
> to specify the lowest svn value that is acceptable.

Reviewer: ```suggestion
or may not be in a security critical condition. The Endorser may provide a minimum svn value using `min-svn`
to specify the lowest svn value that is acceptable.
```

> Objects with svn values that are equal to or greater than min-svn do not signal a secrity critical condition.

Reviewer: ```suggestion
Objects with svn values that are equal to or greater than `min-svn` do not signal a security critical condition.
```

> Objects that are below the min-svn value are in a security critical condition that is unsafe for normal operations.

Reviewer: ```suggestion
Objects that are below the `min-svn` value are in a security critical condition that is unsafe for normal operations.
```
